### PR TITLE
Results Tab

### PR DESCRIPTION
**Pull request by roheunji**
TabView Component completed with basic structure for the results list. I made it so when each tab is clicked on that view is rendered with the list as default. 
Screenshot from 2021-12-10 11-21-59
Screenshot from 2021-12-10 11-22-05
Screenshot from 2021-12-10 11-22-09

Ticket
https://trello.com/c/4b8Zv6yc/45-create-tabs-for-the-different-views

@jnbooth please review at your earliest convenience. Thank you!

### CODE REVIEW

**Comment by roheunji** [comment]
[Create tabs for the different views](https://trello.com/c/4b8Zv6yc/45-create-tabs-for-the-different-views)

File: src/components/search/results/EventItem.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        <div className="event-item-time">
+          Day, Date, @ Time Timezone
+        </div>
+        <div>Title of Event</div>
+        <div>Performer Name</div>
+        <div>Location</div>
+      </div>
+      <div className="eventItemPhoto">
+        <img
+          className={styles.eventItemImage}
```
Comment (jnbooth) [inline src/components/search/results/EventItem.jsx:17]: Nice use of module CSS!

File: src/components/search/results/EventItem.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          src="https://freerangestock.com/thumbnail/115648/music-notes-.jpg"
+          alt="event image"
```
Comment (jnbooth) [inline src/components/search/results/EventItem.jsx:19]: Good accessibility by including an alt tag

File: src/components/search/results/EventList.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import EventItem from './EventItem';
+
+const EventList = () => {
+  return (
+    <div className="event-list-Container">
```
Comment (jnbooth) [inline src/components/search/results/EventList.jsx:6]: You might want to see about refactoring this to use module CSS like the above example

File: src/components/search/results/ResultSection.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import TabViews from './TabViews';
+
+const ResultSection = () => {
+  return (
+    <div className="result-container">
```
Comment (jnbooth) [inline src/components/search/results/ResultSection.jsx:6]: Same as above, unless result-container is from a CSS framework

File: src/components/search/results/TabViews.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { BsListUl, BsMap, BsCalendar3Week } from 'react-icons/bs';
+import EventList from './EventList';
+import styles from '../../../styles/resultList.module.css';
+
+const TabViews = () => {
+  const [view, setView] = useState(1);
+  const changeView = (index) => {
+    setView(index);
+  };
```
Comment (jnbooth) [inline src/components/search/results/TabViews.jsx:10]: By the way, this is the same as simply writing `const changeView = setView`. Think about it from a "leading-the-interpreter" mindset. Which means you don't even need `changeView` at all, just `setView`.

File: src/components/search/results/TabViews.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+          <BsListUl />
+        </button>
+        <button
+          className={view === 2 ? styles.activeTabs : styles.tabs}
+          onClick={() => changeView(2)}>
+          <BsMap />
+        </button>
+        <button
+          className={view === 3 ? styles.activeTabs : styles.tabs}
+          onClick={() => changeView(3)}>
```
Comment (jnbooth) [inline src/components/search/results/TabViews.jsx:27]: There's a lot of repetitive code here that you could simplify down to something that looks like this:
```jsx
<TabContainer onSelect={setView}>
  <BsListUl />
  <BsMap />
  <BsCalendar3Week />
</TabContainer>
```
But you would need to create `TabContainer` yourself, unless some other library has already made one. Something to think about.

File: src/components/search/results/TabViews.jsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+          <BsCalendar3Week />
+        </button>
+      </div>
+      <div className={styles.tabsContent}>
+        <div className={view === 1 ? styles.activeContent : styles.content}>
+          {view === 1 ? <EventList /> : <></>}
```
Comment (jnbooth) [inline src/components/search/results/TabViews.jsx:33]: Probably better to use `null` instead of `<></>`.

**Review by jnbooth — APPROVED** [review]

**Pull request merged**